### PR TITLE
Use bin-runtime in compilation and post-deploy bytecode check

### PR DESCRIPTION
**Pull request by loredanacirstea**
fixes https://github.com/raiden-network/raiden-contracts/issues/419

Adds `bin-runtime` computed by `solc` to the precompiled data.
Uses `bin-runtime` from the precompiled data to check the post-deployment bytecode.

### CODE REVIEW

File: raiden_contracts/data/contracts.json [hunk 1]
```diff
@@ -1 +1 @@
..._hash\",\"type\":\"bytes32\"},{\"name\":\"nonce\",\"type\":\"uint256\"},{\"name\":\"additional_hash\",\"type\":\"bytes32\"},{\"name\":\"closing_signature\",\"type\":\"bytes\"},{\"name\":\"non_closing_signature\",\"type\":\"bytes\"}],\"name\":\"updateNonClosingBalanceProof\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"constant\":true,\"inputs\":[],\"name\":\"token\",\"outputs\":[{\"name\":\"\",\"type\":\"address\"}],\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"},{\"constant\":true,\"inputs\":[{\"name\":\"channel_identifier\",\"type\":\"uint256\"},{\"name\":\"participant\",\"type\":\"address\"},{\"name\":\"partner\",\"type\":\"address\"}],\"name\":\"getUnlockIdentifier\",\"outputs\":[{\"name\":\"\",\"type\":\"bytes32\"}],\"payable\":false,\"stateMutability\":\"pure\",\"type\":\"function\"},{\"inputs\":[{\"name\":\"_token_address\",\"type\":\"address\"},{\"name\":\"_secret_registry\",\"type\":\"address\"},{\"name\":\"_chain_id\",\"type\":\"uint256\"},{\"name\":\"_settlement_timeout_min\",\"type\":\"uint256\"},{\"name\":\"_settlement_timeout_max\",\"type\":\"uint256\"},{\"name\":\"_deprecation_executor\",\"type\":
```
Comment (pirapira) [inline raiden_contracts/data/contracts.json]: Yes, I've seen a new `bin-runtime` here.

File: raiden_contracts/utils/bytecode.py [hunk 2]
```diff
@@ -5,7 +5,6 @@ def runtime_hexcode(contracts_manager, name, length):
         name: name of the contract such as CONTRACT_TOKEN_NETWORK
         length: the length of the runtime code
     """
-    compiled_bytecode = contracts_manager.contracts[name]['bin']
-    compiled_bytecode = compiled_bytecode[-length:]
+    compiled_bytecode = contracts_manager.contracts[name]['bin-runtime']
     compiled_bytecode = hex(int(compiled_bytecode, 16))
```
Comment (pirapira) [inline raiden_contracts/utils/bytecode.py]: I believe this `hex(int(...))` line can also be removed.

Comment (pirapira) [reply raiden_contracts/utils/bytecode.py]: Or not.  Maybe the initial `0x` is missing or so.

Comment (loredanacirstea) [reply raiden_contracts/utils/bytecode.py]: `web3.eth.getCode(address)` returns `HexBytes('0x..')`, so you would still have to cast.

**Pull request merged**